### PR TITLE
OSD-21166: Add make coverage target

### PR DESCRIPTION
**Pull request by samanthajayasinghe**
# What
Adding coverage target to generate codecov report

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/openshift/backplane-tools
 
-go 1.20
+go 1.21
+
+toolchain go1.22.0
```
Comment (typeid) [inline go.mod]: Can we do that hop without further consideration? I believe we need to simulatenously upgrade the /release repo pipeline image: https://github.com/openshift/release/blob/959b5e409ebcd84fe477b54c31e96ee9bfd53322/ci-operator/config/openshift/backplane-tools/openshift-backplane-tools-main.yaml#L6

Comment (samanthajayasinghe) [reply go.mod]: Thanks @typeid , I updated the release PR with go 1.21 image

Comment (samanthajayasinghe) [reply go.mod]: All good . I reverted the go version changes and we can merge this PR first

**Review by typeid — APPROVED** [review]

**Pull request merged**